### PR TITLE
Move data preprocessing to a separate script

### PR DESCRIPTION
**Pull request by AmevinLS**

### CODE REVIEW

File: scripts/data_prep.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def preprocess_dataset(
+    orig_df: pd.DataFrame, test_size: float
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    logging.info("Cleaning data...")
+    clean_df = clean_dataset(orig_df)
+
+    logging.info("Train/test splitting data...")
+    train_df, test_df = train_test_split(clean_df, test_size=test_size, random_state=42)
```
Comment (ekohachi22) [inline scripts/data_prep.py]: Maybe we should also add `random_state` as a parameter, so that we can control the seed globally from e,g. some config file

**Review by ekohachi22 — APPROVED** [review]
